crates/mcp-client/src/client.rs

```diff
@@ -49,47 +51,17 @@ pub struct InitializeParams {
     pub client_info: ClientInfo,
 }
 
-/// The MCP client trait defining the interface for MCP operations.
-#[async_trait::async_trait]
-pub trait McpClient: Send + Sync + 'static {
-    /// Initialize the connection with the server.
-    async fn initialize(
-        &self,
-        info: ClientInfo,
-        capabilities: ClientCapabilities,
-    ) -> Result<InitializeResult, Error>;
-
-    /// List available resources.
-    async fn list_resources(&self) -> Result<ListResourcesResult, Error>;
-
-    /// Read a resource's content.
-    async fn read_resource(&self, uri: &str) -> Result<ReadResourceResult, Error>;
-
-    /// List available tools.
-    async fn list_tools(&self) -> Result<ListToolsResult, Error>;
-
-    /// Call a specific tool with arguments.
-    async fn call_tool(&self, name: &str, arguments: Value) -> Result<CallToolResult, Error>;
-}
-
-/// Standard implementation of the MCP client that sends requests via the provided service.
-pub struct McpClientImpl<S> {
-    service: Mutex<S>,
+/// The MCP client is the interface for MCP operations.
+pub struct McpClient {
+    service: Mutex<TransportHandle>,
```

Comment:
nit/can address later: does this being a concrete TransportHandle instead of a generic service preclude middleware, if we wanted to add that sometime? e.g. Timeout middleware. I don't immediately see why this couldn't continue to use a generic but maybe it complicates something

Reply:
alternatively, we can make TransportHandle a trait and then have the function for both stdio and sse